### ISSUE TITLE
Clean installation from Recovery ignores drive selection, always installs on same drive as recovery

### ISSUE THREAD

**Issue by spxak1** [pop-os/pop#2469]
**Distribution (run `cat /etc/os-release`):**
 NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os

 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

**Issue/Bug Description:**
Rebooting to recovery and performing a clean install ignores the drive selection. Instead, it always installs on the drive where the recovery is located.

So if you have your installation on one drive, connect a second drive, reboot to recovery, and attempt a clean install on the second drive (by selecting it from the prompt), the installation will be performed on the first drive. This will wipe your installation, and obviously leave the second drive blank, but it's obvious the former is the big issue here.

**Steps to reproduce (if you know):**
   
* Clean install from USB to nvme drive
* Reboot to new installation
* Reboot to recovery
* Follow prompts to clean install
* Select SATA drive
* Installation proceeds on nvme drive instead.

### Further testing:

* Reboot to USB. Currently nvme drive installed with Pop
* Select clean install
* Select SATA drive
* Installation proceeds as expected on SATA drive.

### Further testing:

* Reboot to SATA drive's Pop installation.
* Reboot to SATA drive's recovery
* Follow prompts to clean install
* Select NVME drive as installation target
* Installation proceeds on the SATA drive, again, not the selected drive

**Expected behavior:**
Installation should be performed on the selected drive. 

**Other Notes:**

This issue was raised and discussed here: https://www.reddit.com/r/pop_os/comments/v9i8fc/linux_newbie_maybe_found_serious_bug_when/

The OP has a short video as well. But this is easy to reproduce.

**Comment by pbui** [comment]
I believe the issue is due to the logic here:  https://github.com/pop-os/installer/blob/master/src/Views/DiskView.vala#L73-L86

If `install_options.has_recovery ()` is true then the `recovery` disk is always used even if another disk is chosen.

It is not clear what the solution should be.  One possible solution is remove the `if` statement and always use the selected disk (ie. the `else` block).  Another option could be to not present anything but the original disk if in recovery mode.

I think either would make sense... and should be straightforward to implement.  Currently, the presentation is confusing because a clean install can wipe out the original disk instead of the one chosen.  So either allow the user to actually clean install to another disk or disable that optional.

**Comment by pbui** [comment]
@spxak1 It might make more sense to have this issue on the https://github.com/pop-os/installer repository... but I'm guessing someone on the pop-os team can move it there.

**Comment by spxak1** [comment]
> @spxak1 It might make more sense to have this issue on the https://github.com/pop-os/installer repository... but I'm guessing someone on the pop-os team can move it there.

Oh missed that. You're right. I'll wait to see if it's moved or if it grabs any attention before I repost. Thanks.

**Comment by spxak1** [comment]
> I believe the issue is due to the logic here: https://github.com/pop-os/installer/blob/master/src/Views/DiskView.vala#L73-L86
> 
> If `install_options.has_recovery ()` is true then the `recovery` disk is always used even if another disk is chosen.
> 
> It is not clear what the solution should be. One possible solution is remove the `if` statement and always use the selected disk (ie. the `else` block). Another option could be to not present anything but the original disk if in recovery mode.
> 
> I think either would make sense... and should be straightforward to implement. Currently, the presentation is confusing because a clean install can wipe out the original disk instead of the one chosen. So either allow the user to actually clean install to another disk or disable that optional.

Yes, it looks more like a "feature" than a bug. However since users are offered the choice, and that choice is not taken into considerations, users end up with the wrong drive formatted. 

One would argue that booting to recovery is (should be) about saving an existing installation, so a clear install on the existing drive "makes sense". I don't argue with that, but it has to be clear what is happening, what drive is about to be partitioned and formatted and what data is going to be lost.

On a personal preference not, I don't see why the recovery can't offer what the USB can. In other words, the recovery offers the options expected, and a clean install on the device of choice. 

There is no reason why one wouldn't want to do a clean install on a second drive (to move to another system maybe) without the need to create a new USB for it, since recovery is there, and the option is available (but not currently working).

But that's for the team to decide. 

At this point I think it's only right for the user to know what drive is going to be losing data. Thanks.
